File: app/src/main/java/com/wj/mydemo/BaseActivity.java
package com.wj.mydemo;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.gyf.immersionbar.ImmersionBar;
import com.tbruyelle.rxpermissions2.RxPermissions;


/**
 * Created by Administrator on 2018/1/4.
 */

public abstract class BaseActivity extends AppCompatActivity {


    protected RxPermissions rxPermissions;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(setLayoutId());
        ImmersionBar.with(this).init();
        rxPermissions = new RxPermissions(this);
        initView();
        initData();
        setEvent();
    }

    protected abstract int setLayoutId();

    protected abstract void initView();

    protected abstract void initData();

    protected abstract void setEvent();

}

File: app/src/main/java/com/wj/mydemo/BaseActivity.kt
package com.wj.mydemo

import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import butterknife.ButterKnife

import com.gyf.immersionbar.ImmersionBar
import com.tbruyelle.rxpermissions2.RxPermissions

/**
 * Created by Administrator on 2018/1/4.
 */

abstract class BaseActivity : AppCompatActivity() {
    protected var rxPermissions: RxPermissions? = null
    protected var barView: View? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(setLayoutId())
        ButterKnife.bind(this)
        ImmersionBar
            .with(this)
            .navigationBarColorTransform(R.color.blue_primary)
            //                .navigationBarDarkIcon(true) //导航栏图标是深色，不写默认为亮色
            //                .fullScreen(true)
            .navigationBarAlpha(1.0f)  //导航栏透明度，不写默认0.0F
            .init()
        rxPermissions = RxPermissions(this)
        barView = findViewById(R.id.barView)
        ImmersionBar.setStatusBarView(this, barView)
        initView()
        initData()
        setEvent()
    }

    protected abstract fun setLayoutId(): Int

    protected open fun initView() {}

    protected abstract fun initData()

    protected abstract fun setEvent()

}